### ISSUE TITLE
Differences in free nats clipping between Dreamer, early and final PlaNet implementation

### ISSUE THREAD

**Issue by TyPh00nCdr** [danijar/dreamer#45]
In PlaNet you changed the calculation of the free nats between the latest two commits from ([5bb34f9](https://github.com/google-research/planet/blob/5bb34f92edf549db342c7d923da392d13559c79d/planet/training/utility.py#L195))
```
loss = tf.maximum(tf.cast(free_nats, tf.float32), loss)
```
to ([cbe77fc](https://github.com/google-research/planet/blob/cbe77fc011299becf6c3805d6007c5bf58012f87/planet/training/utility.py#L211))
```
loss = tf.maximum(0.0, loss - float(free_nats))
```
Now in Dreamer you seem to have reverted back to the first version, again
https://github.com/danijar/dreamer/blob/2d58db947b05ef1fba57dbc18cd7c2b774136ee5/dreamer.py#L175
As I see it, the two versions should be equivalent to 

1. `loss = tf.clip_by_value(loss, clip_value_min=free_nats)`
2. `loss = tf.clip_by_value(loss, clip_value_min=free_nats) - free_nats`

if I'm not mistaken. So one will always clip the KL-loss to at least the value of the free nats, while the other will always reduce the loss by the value of the free nats (without making the loss negative)?

Which one is the preferred approach to free nats clipping?

**Comment by danijar** [comment]
The two have versions have the same gradient, so the choice does not influence the model training. The only difference is what number is logged as the loss so it's up to personal preference.

**Comment by TyPh00nCdr** [comment]
Sure, that explains it! My bad. Thanks!
